src/hypergol/tensorflow_model_manager.py

```diff
@@ -121,20 +128,19 @@ def run(self, stepCount, evaluationSteps, tensorboardSteps, metadataSteps, train
             num total steps in schedule
         evaluationSteps: List[int]
             which steps to produce metrics on an evaluation sample
-        tensorboardSteps: List[int]
-            which steps to log taining loss to tensorboard
         metadataSteps: List[int]
             which steps to log metadata to tensorboard
-        trainingSteps: List[int]
-            which steps to train model
         """
-        if trainingSteps is None:
-            trainingSteps = range(stepCount)
-        self.initialize()
-        for k in tqdm(range(stepCount)):
-            if k in trainingSteps:
-                self.train(withLogging=k in tensorboardSteps, withMetadata=k in metadataSteps)
-            if k in evaluationSteps:
-                self.save_model()
-                self.evaluate(withLogging=k in tensorboardSteps, withMetadata=k in metadataSteps)
-        self.save_model()
+        self.start()
+        try:
```

Comment:
This try seems a little big, is this necessary?

Reply:
Yes, training usually stopped with a ctrl-c and we need to close the dataset in the batchprocessor or it cannot be used at all (chk file is not created). now that you said this maybe the self.save_model needs to be in the finally as well.